k8s-deployer/src/main/java/io/hyperfoil/deploy/k8s/K8sDeployer.java

```diff
@@ -306,25 +306,26 @@ public boolean hasControllerLog() {
    }
 
    @Override
-   public void downloadControllerLog(long offset, String destinationFile, Handler<AsyncResult<Void>> handler) {
-      downloadRunningLog(CONTROLLER_POD_NAME, offset, destinationFile, handler);
+   public void downloadControllerLog(long offset, long maxLength, String destinationFile, Handler<AsyncResult<Void>> handler) {
+      downloadRunningLog(CONTROLLER_POD_NAME, offset, maxLength, destinationFile, handler);
    }
 
    @Override
-   public void downloadAgentLog(DeployedAgent deployedAgent, long offset, String destinationFile,
+   public void downloadAgentLog(DeployedAgent deployedAgent, long offset, long maxLength, String destinationFile,
          Handler<AsyncResult<Void>> handler) {
       K8sAgent agent = (K8sAgent) deployedAgent;
       ensureClient();
       if (agent.outputPath != null) {
          try (InputStream stream = new FileInputStream(agent.outputPath.toFile())) {
             skipBytes(offset, stream);
-            Files.copy(stream, Paths.get(destinationFile), StandardCopyOption.REPLACE_EXISTING);
+            Files.copy(new TruncatingInputStream(stream, maxLength), Paths.get(destinationFile),
```

Comment:
Similarly to before: can you add a log in case the copied data is less than the existing file size?